File: Backend/DTOs/StudentDto.cs
namespace Backend.DTOs
{
    /// <summary>A data transfer object for a student.</summary>
    public class StudentDto : DomainUserDto
    {
        public int EntranceYear { get; set; } // might be redundant

        public DepartmentInfoDto Major { get; set; }

        public ICollection<DepartmentInfoDto> Minors { get; set; }

        public double CGPA { get; set; }

        public double ExchangeScore { get; set; }

        public SemesterInfoDto PreferredSemester { get; set; }

        public ICollection<string> PreferredSchools { get; set; }

        public string ExchangeSchool { get; set; }

        public ICollection<CTEFormDto> CTEForms { get; set; }

        public ICollection<PreApprovalFormDto> PreApprovalForms { get; set; }

        public ICollection<EquivalenceRequestDto> EquivalenceRequestForms { get; set; }

        public ICollection<ToDoItemDto> ToDoList { get; set; }
    }
}

File: Frontend/src/dtos/student.dto.ts
import {DepartmentInfoDto} from "./department-info.dto";
import {SemesterInfoDto} from "./semester-info.dto";
import {CteFormDto} from "./cte-form.dto";
import {PreApprovalFormDto} from "./pre-approval-form.dto";
import {EquivalanceRequestDto} from "./equivalance-request.dto";

export class StudentDto {
  entranceYear: number;
  major: DepartmentInfoDto;
  minors: ArrayLike<DepartmentInfoDto>;
  cgpa: number;
  exchangeScore: number;
  preferredSemester: SemesterInfoDto;
  preferredSchools: ArrayLike<string>;
  exchangeSchool: string;
  cteForms: ArrayLike<CteFormDto>;
  preApprovalForms: ArrayLike<PreApprovalFormDto>;
  equivalenceRequestForms: ArrayLike<EquivalanceRequestDto>;
}
